src/download.rs

```diff
@@ -349,9 +303,90 @@ pub fn download_to_tempfile(url: &str) -> Result<File> {
     Ok(f)
 }
 
-/// Format a size in bytes.
-fn format_bytes(count: u64) -> String {
-    Byte::from_bytes(count.into())
-        .get_appropriate_unit(true)
-        .format(1)
+struct ProgressReader<'a, R: Read> {
+    source: R,
+    length: Option<u64>,
+    artifact_type: &'a str,
+
+    position: u64,
+    last_report: Instant,
+
+    tty: bool,
+    prologue: &'static str,
+    epilogue: &'static str,
+}
+
+impl<'a, R: Read> ProgressReader<'a, R> {
+    fn new(source: R, length: Option<u64>, artifact_type: &'a str) -> Result<Self> {
+        let tty = isatty(stderr().as_raw_fd()).chain_err(|| "checking if stderr is a TTY")?;
```

Comment:
Nit: I think it is more useful to log a warning and assume `false` if this `isatty` check fails.
That is, assuming this logic is mostly cosmetic, I think we are more interested into having the download logic still working (just with less bells and whistles) on those corner cases.